torchtune/dev/grpo/types.py

```diff
@@ -5,9 +5,203 @@
 # LICENSE file in the root directory of this source tree.
 
 
-from typing import NamedTuple, Optional
-
+from dataclasses import dataclass
 import torch
+from typing import List, Dict, Optional, Any
+
+@dataclass
+class UnscoredTrajectory:
+    """
+    Represents a single trajectory generated by the policy model before scoring.
+
+    Attributes:
+        prompt_tokens (torch.Tensor): Unpadded prompt tokens. Shape: (prompt_len,).
+        response_tokens (torch.Tensor): Unpadded response tokens generated by the policy.
+            Shape: (response_len,).
+        prompt_len (int): Length of the prompt tokens.
+        response_len (int): Length of the response tokens.
+        logprobs (torch.Tensor): Log probabilities of the response tokens from the
+            generating policy (e.g., vLLM). Shape: (response_len,).
+        answer (str): The ground truth or reference answer associated with the prompt.
+        sequence_id (str): A unique identifier for this specific trajectory.
+        policy_version (int): The version identifier of the policy model that generated
+            this trajectory.
+        group_id (str): An identifier grouping trajectories generated from the same base
+            prompt, used for advantage computation.
+    """
+    prompt_tokens: torch.Tensor
+    response_tokens: torch.Tensor
+    prompt_len: int
+    response_len: int
+    logprobs: torch.Tensor
+    answer: str
+    sequence_id: str
+    policy_version: int
+    group_id: str
+
+@dataclass
+class GroupedUnscoredTrajectories:
+    """
+    Represents a collection of unscored trajectories generated from the same base prompt.
+
+    Attributes:
+        trajectories (List[UnscoredTrajectory]): A list of individual unscored trajectories
+            belonging to this group.
+        group_id (str): The unique identifier shared by all trajectories in this group.
+        total_tokens (int): The sum of prompt lengths and response lengths for all
+            trajectories in the group.
+    """
+    trajectories: List[UnscoredTrajectory]
```

Comment:
Yeah, let's not nest. It's just too annoying to deal with. Inheriting dataclasses is also a bit icky, but at least it's easier to work with as fields aren't nested